### PR TITLE
Developer Integration Guide: SDK Setup 2.0.0

### PR DESCRIPTION
**Pull request by Will5**
An extension of #205 that incorporates the SDK 2.0.0 release

### CODE REVIEW

File: docs/sdks/flutter-sdk/overview.mdx
Comment (turnipdabeets) [inline docs/sdks/flutter-sdk/overview.mdx:14]: Oh I see, you're importing everything over (in response to the question in my bind PR). It might be easier to maintain if we kept just the one SDK Setup file?

Comment (Will5) [reply docs/sdks/flutter-sdk/overview.mdx:14]: I agree if we're ditching these overview files now. This makes it easier to maintain both.

File: docs/workflows/_sdk-setup/_authentication-swift.mdx
Comment (turnipdabeets) [inline docs/workflows/_sdk-setup/_authentication-swift.mdx:10]: for iOS this is just `id` because the Type is `Passkey.Id` so you can't pass any other id except a `Passkey.Id` :)
```suggestion
    id: Passkey.Id,
```

File: docs/workflows/_sdk-setup/_authentication-react-native.mdx
Comment (turnipdabeets) [inline docs/workflows/_sdk-setup/_authentication-react-native.mdx:18]:
```suggestion
  redirectUrl: string;
```

File: docs/workflows/_sdk-setup/_passkey-management/_deleting-a-passkey-swift.mdx
Comment (turnipdabeets) [inline docs/workflows/_sdk-setup/_passkey-management/_deleting-a-passkey-swift.mdx:5]:
```suggestion
    for id: Passkey.Id,
```

File: docs/sdks/flutter-sdk/overview.mdx
Comment (turnipdabeets) [reply docs/sdks/flutter-sdk/overview.mdx:14]: Do you want to go ahead and rejoin these so we have one file for now?

File: sidebars.js
Comment (turnipdabeets) [inline sidebars.js:88]: I think you can remove all the auto-generated doc links here if you added all of these SDK set up. So ideally the SDK folder would only have these files and we remove the nested individual SDK folders:
```
SDK Setup
Bind Passkey To User
Authenticate with Passkey
Integrate with Auth0 
Integrate with Okta
```
(later we can join the okta and auth0 guides into one file too will different code snippets)

Comment (Will5) [reply sidebars.js]: Done 👍

File: docs/workflows/sdk-setup.mdx
Comment (turnipdabeets) [inline docs/workflows/sdk-setup.mdx:140]: Maybe for now make these todo links so we can link to the other guides when they merge and you can remove the overviews

Comment (Will5) [reply docs/workflows/sdk-setup.mdx:140]: Yeah, I was just using these as placeholders 🙂 I figured since we had existing content, I might as well link to them, until the newer content is merged

File: docs/workflows/sdk-setup.mdx
Comment (turnipdabeets) [inline docs/workflows/sdk-setup.mdx:127]:
```suggestion
- [Documentation](https://gobeyondidentity.github.io/bi-sdk-swift/documentation/beyondidentityembedded/)
```

File: docs/sdks/flutter-sdk/integrate-with-auth0.md
Comment (turnipdabeets) [inline docs/sdks/flutter-sdk/integrate-with-auth0.md:13]: I think you lost most of the 2.0.0 functions like authenticate somehow?

File: docs/workflows/sdk-setup.mdx
Comment (Will5) [reply docs/workflows/sdk-setup.mdx:127]: Huh, I thought I grabbed that 🤔 Thanks!

File: docs/workflows/_sdk-setup/_passkey-management/_deleting-a-passkey-flutter.mdx
Comment (turnipdabeets) [inline docs/workflows/_sdk-setup/_passkey-management/_deleting-a-passkey-flutter.mdx:1]: nit: the text for the various functions all various functions seems to be repeated . You can probably move that to right under the header - in this case `### Deleting a Passkey`

File: docs/workflows/sdk-setup.mdx
Comment (turnipdabeets) [inline docs/workflows/sdk-setup.mdx:69]: This also works without the string
```suggestion
<Tabs groupId="sdks" queryString>
```

File: package.json
Comment (turnipdabeets) [inline package.json:7]: thanks for this!

File: sidebars.js
Comment (turnipdabeets) [inline sidebars.js:57]: It seems like I was wrong and we might be keeping the individual folders after all. Jen's making individual reference documents. So it might make sense to leave these here. I think it's fine to keep them linked in the set up overview too.

Comment (Will5) [reply sidebars.js:57]: Ok, I didn't think we were removing the individual folders yet anyway. I can put these back.

File: package.json
Comment (Will5) [reply package.json:7]: np! I wanted to be able to link to the sdk page to a specific tab, which was literally just released! 😄

File: docs/workflows/sdk-setup.mdx
Comment (Will5) [reply docs/workflows/sdk-setup.mdx:69]: Interesting! I'll test locally and remove

File: docs/workflows/_sdk-setup/_passkey-management/_deleting-a-passkey-flutter.mdx
Comment (Will5) [reply docs/workflows/_sdk-setup/_passkey-management/_deleting-a-passkey-flutter.mdx:1]: I'm debating this. I see I can do that with "Bind Passkey URL Validation", "Authenticate URL Validation",
"Listing Passkeys" and "Deleting a Passkey", but I can't do that with "Setup", "Binding a Passkey" or "Authentication", at least as they're currently written. I might at least wait until further review.

File: docs/workflows/sdk-setup.mdx
Comment (turnipdabeets) [inline docs/workflows/sdk-setup.mdx:202]: I wonder if this should be its own section like `Sample App` at the top? Maybe the section could be called something like `SDK API Documentation`

Comment (Will5) [reply docs/workflows/sdk-setup.mdx:202]: Do you mean move this section underneath "Sample App" and rename it to "SDK API Documentation"? I did consider it there, but I thought it would be better to walk developers through the basic setup before showing them all of the other functions. What do you think?

I do like "SDK API Documentation" as a header better.

Comment (turnipdabeets) [reply docs/workflows/sdk-setup.mdx:202]: I was thinking it could have it's own section like how "Sample App" has it's own section and header.

Comment (turnipdabeets) [reply docs/workflows/sdk-setup.mdx:202]: Oh I see you changed it! Yea like that but with the icons too maybe and some text like the sample app section

File: docs/workflows/_sdk-setup/_passkey-management/_deleting-a-passkey-flutter.mdx
Comment (turnipdabeets) [reply docs/workflows/_sdk-setup/_passkey-management/_deleting-a-passkey-flutter.mdx:1]: It looks like we should be able to do this for "Binding a Passkey" and "Authentication". There are some differences written under JS but the info should be the same info for all the SDKs when calling these functions so we can consolidate the differences. If we remove the text inside these then the code snippets can be reused in other documents like possibly the ones Jen is writing. It'll also make it easier to make updates to text if it's only in one place.

Comment (Will5) [reply docs/workflows/_sdk-setup/_passkey-management/_deleting-a-passkey-flutter.mdx]: I agree that reusable code snippet files would be ideal, but I am not sure yet if these should be those files. I suppose they can be, but a couple of these contain both the request and the response, which I think I should split out first.

However, I notice we still have some inconsistencies. For example, `BindPasskeyResponse` has a url which depending on the sdk is either `postBindRedirect` (JavaScript), `postBindingRedirectUri` (Android, Flutter) or `postBindingRedirectURI` (Swift, React Native). I wonder if this (any/all differences in requests/responses) should be one more item for 2.0.0 before we release.

Comment (Will5) [reply docs/workflows/_sdk-setup/_passkey-management/_deleting-a-passkey-flutter.mdx]: Ok, this should be done.

**Comment by turnipdabeets** [comment]
I think we just need to update this doc to url camelCase `Url`,  `Uri`,  `Id` for the examples and it should be good to go

**Review by sublimezeal — COMMENTED** [review]
I can see that we are in an interim place right now and have gone back and forth on where the info under the SDK's should go, for example. I think it is fine to be here for now as we push the new content and the new articles become available to link to.
My only request regarding this file is to update the Overview at the very top so say "This guide enables you to..." or similar to describe what the reader can expect from this guide right now. It might be "This guide enables you to set up each of the Beyond Identity SDKs and to learn about the functionality  they contain" or similar.

File: docs/sdks/swift-sdk/integrate-with-auth0.md
Comment (turnipdabeets) [inline docs/sdks/swift-sdk/integrate-with-auth0.md:54]:
```suggestion
        id: passkeyId
```

File: docs/sdks/swift-sdk/integrate-with-auth0.md
Comment (turnipdabeets) [inline docs/sdks/swift-sdk/integrate-with-auth0.md:71]:
```suggestion
    id: passkeyId
```

File: docs/sdks/swift-sdk/integrate-with-auth0.md
Comment (turnipdabeets) [inline docs/sdks/swift-sdk/integrate-with-auth0.md:105]:
```suggestion
            id: selectedID
```

File: docs/sdks/swift-sdk/integrate-with-okta.md
Comment (turnipdabeets) [inline docs/sdks/swift-sdk/integrate-with-okta.md:63]:
```suggestion
        id: passkeyId
```

File: docs/sdks/swift-sdk/integrate-with-okta.md
Comment (turnipdabeets) [inline docs/sdks/swift-sdk/integrate-with-okta.md:80]:
```suggestion
    id: passkeyId
```

File: docs/sdks/swift-sdk/integrate-with-okta.md
Comment (turnipdabeets) [inline docs/sdks/swift-sdk/integrate-with-okta.md:114]:
```suggestion
            id: selectedID
```

**Review by turnipdabeets — CHANGES_REQUESTED** [review]
Couple of changes in swift's 2nd idP docs. FYI I remove the integrate-with-beyondidentity documents in #221.

**Review by turnipdabeets — APPROVED** [review]

**Pull request merged**
